test/service/test_manifest.py

```diff
@@ -784,6 +797,127 @@ def test_full_metadata_missing_fields(self):
+
+                        self.assertEqual(file_names.keys(), set(project_ids))
+                        self.assertEqual([2, 2], list(map(len, file_names.values())))
+                        self.assertEqual([1, 1], list(map(len, map(set, file_names.values()))))
+
+    def test_get_seconds_until_expire(self):
+        """
+        Verify a header with valid x-amz-expiration and LastModified values returns the correct expiration value
+        """
+        with mock.patch('time.time', new=lambda: 1547691253.07010):
+            dt_now = datetime.now(timezone.utc)
+            for hours_last_modified in range(0, 25, 6):
+                with self.subTest(hours_last_modified=hours_last_modified):
+                    hours_till_expiration = (config.manifest_expiration * 24) - hours_last_modified
+                    expire_date = (dt_now + timedelta(hours=hours_till_expiration)).strftime("%a, %d %b %Y %H:%M:%S %Z")
+                    header = {
+                        'x-amz-expiration': f'expiry-date="{expire_date}", rule-id="Test Rule"',
+                        'LastModified': dt_now - timedelta(hours=hours_last_modified)
+                    }
+                    seconds_remaining = ManifestResponse._get_seconds_until_expire(header)
+                    self.assertAlmostEqual(seconds_remaining, 60 * 60 * hours_till_expiration, delta=1)
+
+    def test_get_seconds_until_expire_logs(self):
+        """
+        Verify a header with mismatched x-amz-expiration and LastModified values produces an error log message
+        """
+        with mock.patch('time.time', new=lambda: 1547691253.07010):
+            dt_now = datetime.now(timezone.utc)
+
+            def log_messages_from_get_seconds(expire_date: datetime, last_modified: datetime) -> List[str]:
+                expire_date_string = expire_date.strftime("%a, %d %b %Y %H:%M:%S %Z")
+                header = {
+                    'x-amz-expiration': f'expiry-date="{expire_date_string}", rule-id="Test Rule"',
+                    'LastModified': last_modified
+                }
+                with self.assertLogs(level='ERROR') as logs:
+                    ManifestResponse._get_seconds_until_expire(header)
+                    logger.error('Dummy error message to have at least one error message logged')
+                    return logs.output
+
+            hours_until_expire = 8
+            expire_date = dt_now + timedelta(hours=hours_until_expire)
+            last_modified = dt_now - timedelta(hours=((config.manifest_expiration * 24) - hours_until_expire))
+            msg_prefix = 'ERROR:azul.service.responseobjects.hca_response_v5:x-amz-expiration'
+            # Verify no error message is logged when expire_date and last_modified match
+            logs_output = log_messages_from_get_seconds(expire_date, last_modified)
+            self.assertEqual(set([message.startswith(msg_prefix) for message in logs_output]), {False})
```

Comment:
```suggestion
            self.assertEqual(set(['x-amz-expiration' in message for message in logs_output]), {False})
```

In combination with the `logger` argument to `assertLogs` this is less sensitive to log format changes.

Reply:
I will also apply this style of log verification to the other test case `test_metadata_cache_expiration` where i'm verifying the presence and absence of log messages.